ha/core/controllers/pcs/cluster_controller.py

```diff
@@ -119,19 +134,41 @@ def storageset_list(self) -> dict:
         raise HAUnimplemented("This operation is not implemented.")
 
     @controller_error_handler
-    def add_node(self, nodeid: str = None, descfile: str = None) -> dict:
+    def add_node(self, node_id: str = None, cluster_user: str = None,
+                 cluster_password: str = None, desc_file: str = None) -> dict:
         """
         Add new node to cluster.
-
+        :param cluster_user:
+        :param cluster_password:
+        :param node_id:
+        :param desc_file:
         Args:
-            nodeid (str, optional): Provide nodeid. Defaults to None.
-            filename (str, optional): Provide descfile. Defaults to None.
+            node_id (str, optional): Provide node_id. Defaults to None.
+            filename (str, optional): Provide desc_file. Defaults to None.
```

Comment:
The description file handling should be done at upper layer. The cluster management functions should be implemented using standard parameters.

Reply:
Updated

Reply:
Done as per discussion removed descfile parameter and made nodeid, cluster_username & cluster_password as mandatory one